src/Database/Schema/SchemaDialect.php

```diff
@@ -355,6 +355,25 @@ abstract public function indexSql(TableSchema $schema, string $name): string;
      */
     abstract public function truncateTableSql(TableSchema $schema): array;
 
+    /**
+     * Create a SQL snippet for a column based on the array shape
+     * that `describeColumns()` creates.
+     *
+     * @param array $data The column metadata
+     * @return string Generated SQL fragment for a column
+     */
+    public function columnDefinitionSql(array $column): string
```

Comment:
I don't love this method name, but `columnSql` is already in use, and I can't change the signature on that method as it would break backwards compatibility.